### ISSUE TITLE
Database empty while timing block

### ISSUE THREAD

**Issue by tjwallace** [piotrmurach/rspec-benchmark#10]
### Describe the problem
Follow up from #6. Test data cannot be found in the database when using the `expect {  }.to perform_under()` matcher.

### Steps to reproduce the problem
Sample repo can be found here: https://github.com/tjwallace/rspec-benchmark-bug

```
git clone https://github.com/tjwallace/rspec-benchmark-bug.git
cd rspec-benchmark-bug
bundle install
rake db:migrate
rake spec
```

### Actual behaviour
An `ActiveRecord::RecordNotFound` exception is thrown.

### Expected behaviour
The record should be found.

### Describe your environment
* OS version: OSX 10.11.6
* Ruby version: 2.5.1
* RSpec version: 3.8
* RSpec::Benchmark version: 0.4.0

**Comment by piotrmurach** [comment]
@tjwallace @5minpause I've created a minimal [active record example](https://github.com/piotrmurach/rspec-benchmark/blob/master/examples/activerecord.rb) based on the bug Rails app. This isolates the expectation to only ActiveRecord interation without all the Rails extra baggage to see if records in database get affected when the expectation is run more than once.

The following expectation runs 10 times and finds the record without any problem:

```ruby
expect { Post.find(post.id) }.to perform_under(50).ms.sample(10).times
```
This to me provides a proof that the execution of expectation doesn't impact the database and records it holds. This also implies that something else in Rails test setup maybe clearing the database?

**Comment by tjwallace** [comment]
@piotrmurach I found the issue, it's because [rspec-rails wraps each test in a transaction](https://relishapp.com/rspec/rspec-rails/docs/transactions). If I [change the config value to `false`](https://github.com/tjwallace/rspec-benchmark-bug/commit/fef9ed558fd545fd16369df0b8a7633b5130c316) everything passes. I'm guessing because the test block is run in a subprocess, ActiveRecord uses a different connection from the pool, and that connection isn't in the same transaction so it can't see the records in the database.

I wouldn't call disabling transactions a great fix because it is really useful to use transactions to ensure the database is cleaned up between test runs, but I guess the only other option would be to modify `Benchmark::Perf::ExecutionTime` to not use (or conditionally not use) a subprocess?

**Comment by piotrmurach** [comment]
@tjwallace I went down into a rabbit hole. It turns out that ActiveRecord uses a different connection from the connection pool in child process compared to parent regardless whether transactions are used or not. However, when transactions are used together with a forked process, the database is cleaned right away in the child process when it is spawned before the expectation is run. 

These are the calls I've used to verify behaviour:

```ruby
# in parent
puts ActiveRecord::Base.connection.object_id # => 70240804476440
puts ActiveRecord.connection.select_value("SELECT COUNT(*) FROM posts") # => 1

# in child
puts ActiveRecord::Base.connection.object_id # => 70240806012140
puts ActiveRecord.connection.select_value("SELECT COUNT(*) FROM posts") # => 0
```

To me, this seems like a bug in how RSpec handles database connections and process forking. To confirm the situation, I have switched off `transactional_fixtures` and used rspec callbacks to manually clear the database after the test is done and I got the same result. It appears that if a process is forked the callbacks do not wait for the process to finish and clean database right away.

**Comment by piotrmurach** [comment]
I added an option to allow toggling the behaviour of running measurements in a subprocess https://github.com/piotrmurach/benchmark-perf/commit/048bcd6d526d59f0a6b85ccee514d1ec1278533d I also added configuration to `rspec-benchmark` to make it possible to configure how timed specs are run. By default, I switched off running in subprocess given how the `rspec-rails` works by default. This should prevent confusion among new users. However, this option can always be changed:

```ruby
RSpec::Benchmark.configure do |config|
  config.run_in_subprocess = true
end
```

Thank you @tjwallace & @5minpause for bringing this issue to my attention and investigating the reason. All released with `v0.5.0` and many more additions/improvements.
